src/Microsoft.Azure.WebJobs.Host/Executors/TriggeredFunctionData.cs

```diff
@@ -19,5 +20,10 @@ public class TriggeredFunctionData
         /// The trigger value for a specific triggered function invocation.
         /// </summary>
         public object TriggerValue { get; set; }
+
+        /// <summary>
+        /// Optional handler function for processing the invocation.
+        /// </summary>
+        public Func<Func<Task>, Task> InvokeHandler { get; set; }
```

Comment:
This is the key - the listener when it calls the executor can specify this handler